x-pack/plugins/entity_manager/public/lib/entity_client.test.ts

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EntityClient, EntityLatest } from './entity_client';
+import { coreMock } from '@kbn/core/public/mocks';
+
+const commonEntityFields: EntityLatest = {
+  entity: {
+    last_seen_timestamp: '2023-10-09T00:00:00Z',
+    id: '1',
+    display_name: 'entity_name',
+    definition_id: 'entity_definition_id',
+  } as EntityLatest['entity'],
+};
+
+describe('EntityClient', () => {
+  let entityClient: EntityClient;
+
+  beforeEach(() => {
+    entityClient = new EntityClient(coreMock.createStart());
+  });
+
+  describe('asKqlFilter', () => {
+    it('should return the value when indentity_fields is a single string', () => {
+      const entityLatest: EntityLatest = {
+        entity: {
+          ...commonEntityFields.entity,
+          identity_fields: ['service.name', 'service.environment'],
```

Comment:
This is **two** strings though, does this test pass?

Technically this entity instance would be invalid since there can't be optional fields anymore.

Reply:
Shouldn't we ensure that the function doesn't crash despite of that?

Reply:
Your call!